### PR TITLE
Notification: queue ui.go notification-service

### PR DESCRIPTION
**Pull request by zeripath**
As requested by @lunny in  #9902 this extracts the single commit that moves the ui.go notification-service into a separate queue.

### CODE REVIEW

File: modules/notification/ui/ui.go [hunk 1]
```diff
@@ -29,19 +31,24 @@ var (
 
 // NewNotifier create a new notificationService notifier
 func NewNotifier() base.Notifier {
-	return &notificationService{
-		issueQueue: make(chan issueNotificationOpts, 100),
-	}
+	ns := &notificationService{}
```
Comment (lunny) [inline modules/notification/ui/ui.go:34]: Why not just
```go
return &notificationService{
   issueQueue: queue.CreateQueue("notification-service", ns.handle, issueNotificationOpts{}),
}
```

**Comment by lunny** [comment]
A small nit, otherwise LGTM.

File: modules/notification/ui/ui.go [hunk 2]
```diff
@@ -51,35 +58,39 @@ func (ns *notificationService) NotifyCreateIssueComment(doer *models.User, repo
...
 func (ns *notificationService) NotifyNewIssue(issue *models.Issue) {
-	ns.issueQueue <- issueNotificationOpts{
+	_ = ns.issueQueue.Push(issueNotificationOpts{
 		issueID:              issue.ID,
 		notificationAuthorID: issue.Poster.ID,
-	}
+	})
 }
 
 func (ns *notificationService) NotifyIssueChangeStatus(doer *models.User, issue *models.Issue, actionComment *models.Comment, isClosed bool) {
```
Comment (guillep2k) [inline modules/notification/ui/ui.go:71]: Not related to this PR, but shouldn't at least the actionComment ID be passed through as well?

File: modules/notification/ui/ui.go (hunk 1 above)
Comment (zeripath) [reply modules/notification/ui/ui.go:34]: Where do you get the `ns.handle` from?

File: modules/notification/ui/ui.go (hunk 2 above)
Comment (zeripath) [reply modules/notification/ui/ui.go:71]: I would defer to @lunny on this. I don't really understand the notification service.

**Review by lafriks — APPROVED** [review]

File: modules/notification/ui/ui.go (hunk 2 above)
Comment (lunny) [reply modules/notification/ui/ui.go:71]: @guillep2k `actionComment *models.Comment` we can get the ID from the `actionComment`.

**Pull request merged**